### ISSUE TITLE
[FEAT] Linux support

### ISSUE THREAD

**Issue by Toadfield** [AdamXweb/WhiteSurFirefoxThemeMacOS#35]
I like macos big sur themes so I want to make firefox look like safari,but this theme doesn't work on Linux.

**Comment by AdamXweb** [comment]
I'd recommend trying this theme if you're using linux: https://github.com/vinceliuice/WhiteSur-gtk-theme/tree/master/src/other/firefox

**Comment by Toadfield** [comment]
Does that also work for kde?
And also,I just want to use the safari theme,I don't want to use a whole different gtk theme.

**Comment by AdamXweb** [comment]
You can always copy the folder in the link above into your firefox profile folder to get it working. Otherwise i'm pretty sure you can use the `./tweaks.sh -f` script to install it automatically also

**Comment by rivanfebrian123** [comment]
Hi @AdamXweb Are you the author of "Firefox CSS Theme Store"?

**Comment by Toadfield** [comment]
@AdamXweb Does [this](https://github.com/vinceliuice/WhiteSur-gtk-theme/tree/master/src/other/firefox) look the same as your project?

**Comment by rivanfebrian123** [comment]
@Toadfield Actually yeah. @AdamXweb You gotta learn about licensing and don't forget to add `COPYING` in your repo :D But don't worry tho, I already asked Vince to relicense WhiteSur to MIT which is more permisive

**Comment by Toadfield** [comment]
But it would be cooler if this would work on Linux cause [this](https://github.com/vinceliuice/WhiteSur-gtk-theme/tree/master/src/other/firefox) is more than a firefox theme,but I only need the firefox theme.

**Comment by rivanfebrian123** [comment]
It's usable but doesn't quiet work

**WhiteSurFirefoxThemeMacOS**
![Screenshot from 2021-06-13 05-34-02](https://user-images.githubusercontent.com/6300910/121790649-9ddac980-cc0b-11eb-9fe8-a31c0e312fb0.png)


**WhiteSur GTK Theme**
![Screenshot from 2021-06-13 05-35-13](https://user-images.githubusercontent.com/6300910/121790657-ad5a1280-cc0b-11eb-8e18-1153acb26b06.png)


I think we need some query to detect the OS so we can make this theme more portable. References:

- https://codepen.io/jimmykup/pen/ExarGPG
- https://stackoverflow.com/questions/8493589/is-there-a-css-media-query-to-detect-windows

**Comment by AdamXweb** [comment]
> Hi @AdamXweb Are you the author of "Firefox CSS Theme Store"?

Nope, I just added the theme with a pull request into their main repo



> But it would be cooler if this would work on Linux cause [this](https://github.com/vinceliuice/WhiteSur-gtk-theme/tree/master/src/other/firefox) is more than a firefox theme,but I only need the firefox theme.

It would be cool, however like what @rivanfebrian123 mentioned with licensing and copying, i've tried to avoid stepping too close to the original, and to make something for PC and Mac users. If @vinceliuice and @rivanfebrian123 are ok with it, i'd be happy to add support for Linux

> @Toadfield Actually yeah. @AdamXweb You gotta learn about licensing and don't forget to add `COPYING` in your repo :D But don't worry tho, I already asked Vince to relicense WhiteSur to MIT which is more permisive

I'll be sure to add it this week!

> It's usable but doesn't quiet work
> 
> 
> I think we need some query to detect the OS so we can make this theme more portable. References:
> 
>     * https://codepen.io/jimmykup/pen/ExarGPG
> 
>     * https://stackoverflow.com/questions/8493589/is-there-a-css-media-query-to-detect-windows

Definitely possible! The theme already has a few windows specific features (using media queries), so adding some for linux shouldn't be too difficult *hopefully*

Also, I should have a few updates later this week, just been a bit preoccupied moving house

**Comment by rivanfebrian123** [comment]
> however like what @rivanfebrian123 mentioned with licensing and copying, i've tried to avoid stepping too close to the original, and to make something for PC and Mac users

Hey, you know what? WhiteSur is now MIT licenced, so distributors like you, don't have to worry about taking some code from the upstream repo.

> Definitely possible! The theme already has a few windows specific features (using media queries), so adding some for linux shouldn't be too difficult hopefully

Just don't worry about those "bloated" things :) Keep your code clean and neat so it wouldn't be that huge and scary. CMIIW

**Comment by AdamXweb** [comment]
For others looking for Linux support, please check this out:
https://github.com/vinceliuice/WhiteSur-firefox-theme

I'm in the process of adding variables to the theme so that it'll be universal and more compatible. Progress can be viewed here until I receive access to the repo to make modifications:
https://github.com/AdamXweb/WhiteSur-firefox-theme/tree/MacOSWindowsSupport
